### ISSUE TITLE
MA0109 shouldn't trigger on `Main(string[] args)`

### ISSUE THREAD

**Issue by Sella-GH** [meziantou/Meziantou.Analyzer#1122]
### Version of the Meziantou.Analyzer NuGet package

3.0.63

### Rule Identifier

MA0109

### Target Framework

net10.0

### C# Language version

C# 14

### Description

A `Main` method serves as an entrypoint and shouldn't be considered with an overload of a `Span<T>` or `Memory<T>`.

### Reproduction Steps

Minimal code:

```c#
public static void Main(string[] args) // Triggers MA0109
{ }
```
